frame_utils.go

```diff
@@ -0,0 +1,75 @@
+package tchannel
```

Comment:
nit: since this only contains a single type, suggest naming the file after the type (e.g., `recording_frame_pool.go` or `checked_frame_pool.go` if the type is renamed as suggested below